### PR TITLE
Playback speed

### LINKED ISSUE #123: variable playback speed
- option to change the playback speed
  - ExoPlayer offers an option for that
- reset speed everytime you change the queue?
- 0.75, 1, 1.25, 1.5
- investigate logging
  - uniqueSecondsListened = seconds of content consumed = e.g. 10 min
  - timeSpentListening = how much time did the user spend on that content = e.g 7 min
- possible future requirement: limit based on content (music, Fra Kaare, ...)

### PR DESCRIPTION
**Pull request by akrol95**
Closes #123

### CODE REVIEW

File: BMM.Core/GuardedActions/TrackOptions/PrepareTrackOptionsAction.cs [hunk 1]
```diff
@@ -46,6 +52,14 @@ public class PrepareTrackOptionsAction :
         private readonly ISleepTimerService _sleepTimerService;
         private readonly IFirebaseRemoteConfig _firebaseRemoteConfig;
         private readonly IAnalytics _analytics;
+        private readonly IMediaPlayer _mediaPlayer;
+
+        private readonly List<decimal> _availablePlaybackSpeed = new()
+        {
+            0.75m,
```
Comment (kkuepper) [inline BMM.Core/GuardedActions/TrackOptions/PrepareTrackOptionsAction.cs:59]: is 1.00m not an option?

File: BMM.Core/Implementations/Analytics/Event.cs [hunk 2]
```diff
@@ -21,5 +21,7 @@ public class Event
         public const string BackgroundActivityRestrictedPopupShown = "Background Activity restricted popup shown";
         public const string SleepTimerOptionsOpened = "Sleep timer options menu opened";
         public const string SleepTimerOptionSelected = "Sleep timer option selected";
+        public const string PlaybackSpeedOptionsOpened = "Playback speed options menu opened";
+        public const string PlaybackSpeedOptionSelected = "Playback speed option selected";
```
Comment (kkuepper) [inline BMM.Core/Implementations/Analytics/Event.cs:25]: is "Playback speed changed" a better name?

File: BMM.Core/Implementations/PlayObserver/IPlayStatistics.cs [hunk 3]
```diff
@@ -27,6 +27,8 @@ public interface IPlayStatistics
         bool IsCurrentQueueSaved { get; set; }
 
         bool IsPlaying { get; }
+        
+        decimal DesiredRate { get; }
```
Comment (kkuepper) [inline BMM.Core/Implementations/PlayObserver/IPlayStatistics.cs:31]: DesiredPlaybackRate?

File: BMM.Core/Translation/en/main.json [hunk 4]
```diff
@@ -246,7 +246,9 @@
     "Minutes": "{0} minutes",
     "Hour": "{0} hour",
     "Hours": "{0} hours",
-    "Disable": "Disable"
+    "Disable": "Disable",
```
Comment (kkuepper) [inline BMM.Core/Translation/en/main.json:249]: What's the disable used for?

Comment (akrol95) [reply BMM.Core/Translation/en/main.json:249]: It's used for "sleep timer" functionality, in this PR I only added a comma at the end.

File: BMM.Core/Implementations/Analytics/Event.cs (hunk 2 above)
Comment (akrol95) [reply BMM.Core/Implementations/Analytics/Event.cs:25]: I agree

File: BMM.Core/GuardedActions/TrackOptions/PrepareTrackOptionsAction.cs (hunk 1 above)
Comment (akrol95) [reply BMM.Core/GuardedActions/TrackOptions/PrepareTrackOptionsAction.cs:59]: Right now the app will show additional "Default" option if user changed playback speed before. This resets speed to 1x.

But we can follow your idea, and get rid of this "Default" option, and always show the same set:

- 0.75x
- 1x
- 1.25x
- 1.5x

regardless if user selected anything before. 

What more convincing to you

Comment (kkuepper) [reply BMM.Core/GuardedActions/TrackOptions/PrepareTrackOptionsAction.cs:59]: 1x option should be called Normal (like YouTube)

**Review by kkuepper — APPROVED** [review]

**Pull request merged**